consensus/safety-rules/src/error.rs

```diff
@@ -41,3 +45,11 @@ impl From<anyhow::Error> for Error {
         }
     }
 }
+
+impl Error {
+    pub fn result<T, E: std::error::Error>(result: Result<T, E>) -> Result<T, Self> {
```

Comment:
this looks weird, does this work?
`impl<T, E: std::error::Error> From<Result<T, E> for Result<T, Error>`

Reply:
Yup, it is used below. The alternative is to do map_err over and over again.

Reply:
Please don't do this. There are better ways to do these sorts of conversions.